packages/core/src/App/Containers/RealAccountSignup/real-account-signup.jsx

```diff
@@ -277,6 +278,33 @@ const RealAccountSignup = ({
 
     const [assessment_decline, setAssessmentDecline] = React.useState(false);
 
+    const analytic_data = React.useMemo(
+        () => ({
+            form_source: document.referrer,
+            form_name: 'real_account_signup_form',
+            landing_company: real_account_signup_target,
+        }),
+        [real_account_signup_target]
+    );
+
+    const trackEvent = React.useCallback(
+        payload => {
+            if (real_account_signup_target === 'maltainvest') return;
+
+            Analytics.trackEvent('ce_real_account_signup_form', {
+                ...analytic_data,
+                ...payload,
+            });
+        },
+        [analytic_data, real_account_signup_target]
+    );
+
+    React.useEffect(() => {
+        if (is_real_acc_signup_on) {
+            trackEvent({ action: 'open' });
+        }
+    }, [is_real_acc_signup_on, trackEvent]);
```

Comment:
I think this `effect` is not necessary, because this will only fire if `is_real_acc_signup_on` is `true`, I think performing this during rendering would have the same result, right? 🤔 

```suggestion
    React.useEffect(() => {
        if (is_real_acc_signup_on) {
            trackEvent({ action: 'open' });
        }
    }, [is_real_acc_signup_on, trackEvent]);
```